enterprise/micronaut/src/org/netbeans/modules/micronaut/db/MicronautDataEndpointGenerator.java

```diff
@@ -94,8 +104,13 @@ public class MicronautDataEndpointGenerator implements CodeActionProvider, Comma
     private static final String OFFSET =  "offset";
     private static final String REPOSITORIES =  "repositories";
     private static final String ENDPOINTS =  "endpoints";
+    private static final String CONTROLLER_ID =  "controllerId";
 
-    private final Gson gson = new Gson();
+    private final Gson gson = new GsonBuilder().registerTypeAdapter(NotifyDescriptor.QuickPick.Item.class, (JsonDeserializer<NotifyDescriptor.QuickPick.Item>) (JsonElement json, Type type, JsonDeserializationContext jdc) -> {
```

Comment:
Maybe this could be promoted to LSP server implementation, into the 'standard' Gson setup ?

Reply:
Could be done later if more than the single use-case appears.